<?xml version="1.0"?>
<robot name="lidar" xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:arg name="prefix" default="" />
  <xacro:arg name="name" default="lidar" />
  <xacro:arg name="model" default="lms10x" />
  <xacro:arg name="rate" default="50" />
  <xacro:arg name="parent_link" default="base_link" />
  <xacro:arg name="xyz" default="0.0 0.0 0.0" />
  <xacro:arg name="rpy" default="0.0 0.0 0.0" />
  <xacro:arg name="mesh_visual" default="false" />
  <xacro:arg name="ros_namespace" default="/" />

  <xacro:property name="rate"
    value="$(arg rate)" />
  <xacro:property name="model"
    value="$(arg model)" />
  <xacro:property name="rate_hz"
    value="$(arg rate)hz" />
  <xacro:property name="specifications_file"
    value="$(find romea_lidar_description)/config/sick_lms1xx_specifications.yaml" />
  <xacro:property name="specifications"
    value="${load_yaml(specifications_file)}" />

  <xacro:property name="minimal_azimut_angle"
    value="${specifications['minimal_azimut_angle']/180.*pi}" />
  <xacro:property name="maximal_azimut_angle"
    value="${specifications['maximal_azimut_angle']/180.*pi}" />
  <xacro:property name="minimal_range"
    value="${specifications['minimal_range']}" />
  <xacro:property name="maximal_range"
    value="${specifications['maximal_range'][model]}" />
  <xacro:property name="samples"
    value="${specifications['samples'][rate_hz]}" />

  <xacro:include filename="$(find romea_lidar_description)/urdf/lidar.xacro" />
  <xacro:lidar_sensor prefix="$(arg prefix)"
    name="$(arg name)"
    type="sick"
    model="lms1xx"
    xyz="$(arg xyz) "
    rpy="$(arg rpy)"
    parent_link="$(arg parent_link)"
    rate="$(arg rate)"
    mesh_visual="$(arg mesh_visual)"
    minimal_azimut_angle="${minimal_azimut_angle}"
    maximal_azimut_angle="${maximal_azimut_angle}"
    minimal_range="${minimal_range}"
    maximal_range="${maximal_range}"
    samples="${samples}"
    ros_namespace="$(arg ros_namespace)" />

</robot>